src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -0,0 +1,96 @@
+@with_kw struct Ellipsoid{T,CO,RT,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::RT = 1
+    φ::TP = nothing
+    θ::TT = nothing
+
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+end
+
+Ellipsoid{T,CO,RT,TP,TT}( e::Ellipsoid{T,CO,RT,TP,TT}; COT = CO,
+            origin::CartesianPoint{T} = e.origin,
+            rotation::SMatrix{3,3,T,9} = e.rotation) where {T,CO<:Union{ClosedPrimitive, OpenPrimitive},RT,TP,TT} =
+    Ellipsoid{T,COT,RT,TP,TT}(e.r, e.φ, e.θ, origin, rotation)
+
+const Sphere{T,CO,TP,TT} = Ellipsoid{T,CO,T,TP,TT}
+const FullSphere{T,CO} = Ellipsoid{T,CO,T,Nothing,Nothing}
+const FullEllipsoid{T,CO} = Ellipsoid{T,CO,NTuple{3,T},Nothing,Nothing}
+
+function Geometry(::Type{T}, ::Type{Ellipsoid}, dict::AbstractDict, input_units::NamedTuple, transformations::Transformations{T}) where {T}
+    r = parse_r_of_primitive(T, dict, input_units.length)
+    φ = nothing
+    θ = nothing
+    e = Ellipsoid{T,ClosedPrimitive,typeof(r),typeof(φ),typeof(θ)}(
+        r = r, 
+        φ = φ, 
+        θ = θ
+    )
+    transform(e, transformations)
+end
+
+function _in(pt::CartesianPoint, s::FullSphere{<:Any, ClosedPrimitive}) 
+    r = hypot(pt.x, pt.y, pt.z)
+    r <= s.r || csg_isapprox_lin(r, s.r) 
+end
+function _in(pt::CartesianPoint, s::FullSphere{<:Any, OpenPrimitive}) 
+    r = hypot(pt.x, pt.y, pt.z)
+    r < s.r
+end
```

Comment:
And non-full `Spheres`?

Reply:
Can in general be build by subtraction of other primitives. 
We can add additional specialized versions for several primitives later on.